src/Shared/SecretsStore.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json.Nodes;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.UserSecrets;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal;
+
+/// <summary>
+/// Adapted from dotnet user-secrets at https://github.com/dotnet/aspnetcore/blob/482730a4c773ee4b3ae9525186d10999c89b556d/src/Tools/dotnet-user-secrets/src/Internal/SecretsStore.cs
+/// </summary>
+internal class SecretsStore
```

Comment:
Since we are putting this in the `Shared` folder, it would be best if we shared it with the other places that manipulate the user secrets:

https://github.com/dotnet/aspire/blob/9dfbea4e3201fc6de8fb42b14d013ac4fca70517/src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs#L164-L216

Maybe in a follow up PR, or log an issue?

Reply:
Logged #4596